request-management-api/request_api/services/documentservice.py

```diff
@@ -51,6 +53,27 @@ def createrequestdocumentversion(self, requestid, documentid, documentschema, us
         else:
             return self.createrawdocumentversion(requestid, documentid, documentschema, userid)
 
+    def copyrequestdocumenttonewlocation(self, newcategory, documentpath): #documentpath is full url including https://
+        # for old documentpath
+        baseurl = 'https://' + os.getenv("OSS_S3_HOST")
+        location = documentpath.split('/')[3:] # bucket and filename
+        bucket = location[0]
+        source = "/".join(location) # /bucket/filename
+        # for new document path, replace category name in path
+        newlocation = location
+        newlocation[3] = newcategory
+        filename = "/".join(newlocation[1:])
+        newdocumentpath = baseurl + '/' + bucket + '/' + filename
+        try:
+            moveresponse = storageservice().copy_file(source, bucket, filename)
+            if moveresponse['ResponseMetadata']['HTTPStatusCode'] == 200:
+                return {"status": "success", 'documentpath': newdocumentpath}
+            else:
+                raise Exception(moveresponse)
+        except Exception as ex:
+            logging.exception(ex)
+            return {"status": "error"}
```

Comment:
Throw an exception if the boto3 client doesn't return a 200 response for copying the file on s3

Reply:
can use raise for status here, will take fewer lines